src/shared/XLinkPrivateFields.c

```diff
@@ -65,13 +66,18 @@ streamId_t getStreamIdByName(xLinkDesc_t* link, const char* name)
 
 streamDesc_t* getStreamById(void* fd, streamId_t id)
 {
-    XLINK_RET_ERR_IF(id == INVALID_STREAM_ID, NULL);
     xLinkDesc_t* link = getLink(fd);
     XLINK_RET_ERR_IF(link == NULL, NULL);
     int stream;
     for (stream = 0; stream < XLINK_MAX_STREAMS; stream++) {
         if (link->availableStreams[stream].id == id) {
-            XLink_sem_wait(&link->availableStreams[stream].sem);
+
+            int rc;
+            while(((rc = XLink_sem_wait(&link->availableStreams[stream].sem)) == -1) && errno == EINTR)
```

Comment:
What are you trying to do here? If you are trying to replicate a mutex...do not. There is little chance you will do it correctly because writing a core OS locking mechanism is extremely difficult.
Instead, use a mutex.
Or build this code into the function `XLink_sem_wait()` itself. Why would this single location of code benefit from this extra spin loop while all the other places in Xlink do not? In addition, `errno` is not reliable here. Why? Because `Xlink_sem_wait()` (which might generate errno=EINTR) calls `XLink_sem_dec_waits()` within itself before returning the `-1`. Therefore, the `errno` is no longer valid to be checked *outside* of `Xlink_sem_wait()`. You would need to check `errno` inside `Xlink_sem_wait()` directly after the os `sem_wait()` call.
This feels out of alignment.

Reply:
> Why would this single location of code benefit from this extra spin loop while all the other places in Xlink do not? 

Seems like they did something wierd there... They actually put the `EINTR` check inside XLink_sem_wait but are also checking it outside...
Anyway, will drop this as it'll get handled with upstream update by itself.

(I guess main intention was to catch EINTR event and to not continue with execution - probably happened in some case somewhere)